src/Andre/SoulsFormats/SoulsFormats/Formats/MSB/MSBE/PartsParam.cs

```diff
@@ -2388,13 +2406,15 @@ public class ConnectCollision : Part
                 [MSBReference(ReferenceType = typeof(Collision))]
                 [NoRenderGroupInheritence()]
                 public string CollisionName { get; set; }
+
                 [IndexProperty]
-                public int CollisionIndex { get; set; }
+                [XmlIgnore]
+                private int CollisionIndex { get; set; }
 
                 /// <summary>
                 /// The map to load when on this collision.
                 /// </summary>
-                public byte[] MapID { get; private set; }
+                public byte[] MapID { get; set; }
```

Comment:
again privacy thing.
I'm guessing this is a convenience for setting to a mapid.
Perhaps we can write in the length-check in the setter method?

Reply:
Main use case is XML serialization as it would fail on this
If this is handled by some kind of setter method, it should be overriding this default setter, and the propertyshould be backed by a private field